projects/ad7616_sdz/common/ad7616_bd.tcl

```diff
@@ -2,87 +2,177 @@
+#               - Default value is 1
+#
+# LEGEND: Serial    - 1
+#         Parallel  - 0
+#
+# NOTE : This switch is a 'hardware' switch. Please rebuild the design if the
+# variable has been changed.
+#     SL5 - mounted - Serial
+#     SL5 - unmounted - Parallel
+#
+##--------------------------------------------------------------
+
+set SER_PAR_N $ad_project_params(SER_PAR_N)
+puts "build parameters: SER_PAR_N: $SER_PAR_N"
 
 # control lines
 
 create_bd_port -dir O rx_cnvst
-create_bd_port -dir O rx_cs_n
 create_bd_port -dir I rx_busy
 
 # instantiation
 
-ad_ip_instance axi_ad7616 axi_ad7616
-ad_ip_parameter axi_ad7616 CONFIG.IF_TYPE $SI_OR_PI
+# dma
 
 ad_ip_instance axi_dmac axi_ad7616_dma
-ad_ip_parameter axi_ad7616_dma CONFIG.DMA_TYPE_SRC 2
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_TYPE_DEST 0
 ad_ip_parameter axi_ad7616_dma CONFIG.CYCLIC 0
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_2D_TRANSFER 0
-ad_ip_parameter axi_ad7616_dma CONFIG.DMA_DATA_WIDTH_SRC 16
 ad_ip_parameter axi_ad7616_dma CONFIG.DMA_DATA_WIDTH_DEST 64
 
-# interface connections
-if {$SI_OR_PI == 0} {
+# axi_pwm_gen
 
-  ad_connect  rx_sclk axi_ad7616/rx_sclk
-  ad_connect  rx_sdo axi_ad7616/rx_sdo
-  ad_connect  rx_sdi axi_ad7616/rx_sdi
-  ad_connect  rx_cs_n axi_ad7616/rx_cs_n
+ad_ip_instance axi_pwm_gen ad7616_pwm_gen
+ad_ip_parameter ad7616_pwm_gen CONFIG.PULSE_0_PERIOD 100
+ad_ip_parameter ad7616_pwm_gen CONFIG.PULSE_0_WIDTH 5
+ad_ip_parameter ad7616_pwm_gen CONFIG.ASYNC_CLK_EN 1
+
+# axi_clkgen
+
+ad_ip_instance axi_clkgen spi_clkgen
+ad_ip_parameter spi_clkgen CONFIG.CLK0_DIV 6
+ad_ip_parameter spi_clkgen CONFIG.VCO_DIV 1
+ad_ip_parameter spi_clkgen CONFIG.VCO_MUL 6
+
+# trigger to BUSY's negative edge
+
+create_bd_cell -type module -reference sync_bits busy_sync
+create_bd_cell -type module -reference ad_edge_detect busy_capture
+set_property -dict [list CONFIG.EDGE 1] [get_bd_cells busy_capture]
+
+ad_connect spi_clk busy_capture/clk
+ad_connect busy_capture/rst GND
+ad_connect spi_clk busy_sync/out_clk
+ad_connect busy_sync/in_bits rx_busy
+ad_connect busy_sync/out_bits busy_capture/signal_in
+
+if {$SER_PAR_N == 1} {
+  create_bd_intf_port -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 ad7616_spi
+
+  source $ad_hdl_dir/library/spi_engine/scripts/spi_engine.tcl
+
+  set data_width    16
+  set async_spi_clk 1
+  set num_cs        1
+  set num_sdi       2
+  set sdi_delay     1
+  set hier_spi_engine spi_ad7616
+
+  spi_engine_create $hier_spi_engine $data_width $async_spi_clk $num_cs $num_sdi $sdi_delay
+
+  ad_ip_parameter axi_ad7616_dma CONFIG.DMA_DATA_WIDTH_SRC 32
+  ad_ip_parameter axi_ad7616_dma CONFIG.DMA_TYPE_SRC 1
+  ad_ip_parameter axi_ad7616_dma CONFIG.SYNC_TRANSFER_START 0
+  ad_ip_parameter axi_ad7616_dma CONFIG.AXI_SLICE_SRC 0
+  ad_ip_parameter axi_ad7616_dma CONFIG.AXI_SLICE_DEST 1
+
+  # interface connections
+
+  ad_connect  $sys_cpu_clk spi_clkgen/clk
+  ad_connect  spi_clk spi_clkgen/clk_0
```

Comment:
After building the parallel interface mode it seems that this connection should be made interface-independent.

Reply:
Fixed the issue, thank you.